src/items.rs

```diff
@@ -2161,18 +2228,20 @@ impl Rewrite for ast::Param {
                 shape,
                 has_multiple_attr_lines,
             )
+            .ok_or_else(|| RewriteError::Unknown)
         } else if is_named_param(self) {
             let param_name = &self
                 .pat
-                .rewrite(context, Shape::legacy(shape.width, shape.indent))?;
+                .rewrite_result(context, Shape::legacy(shape.width, shape.indent))?;
             let mut result = combine_strs_with_missing_comments(
                 context,
                 &param_attrs_result,
                 param_name,
                 span,
                 shape,
                 !has_multiple_attr_lines && !has_doc_comments,
-            )?;
+            )
+            .ok_or_else(|| RewriteError::Unknown)?;
```

Comment:
Might make sense to keep track that we'll eventually want to change the signature for `combine_strs_with_missing_comments` to return `Result<String, RewriteError>`.